sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc

```diff
@@ -0,0 +1,220 @@
+= sycl_ext_oneapi_group_load_store_extended
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+:language: {basebackend@docbook:c++:cpp}
+
+== Introduction
+
+IMPORTANT: This specification is a draft.
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (c) 2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+== Dependencies
+
+This extension is written against the SYCL 2020 specification, Revision 5. All references below to the "core SYCL specification" or to section numbers in the SYCL specification refer to that revision.
+
+This extension also depends on the following other SYCL extensions:
+
+* link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_local_memory.asciidoc[
+  sycl_ext_oneapi_local_memory].
+
+* link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[
+  sycl_ext_oneapi_group_sort].
+
+* link:../experimental/sycl_ext_oneapi_properties.asciidoc[
+  sycl_ext_oneapi_properties].
+
+== Current State
```

Comment:
I think you should replace this section with one that defines replacements for the single value functions, consistent with your other interfaces, e.g.:

```c++
// Load API
template<typename GroupHelper, typename InputIterator, typename OutputIterator,
        std::size_t ElementsPerWorkItem, typename Properties = properties<>>
void load_over_group(GroupHelper gh, InputIteratorT in_ptr,
                      OutputIteratorT out_ptr, Properties = {});
```

Reply:
Replaced for single values, please, take a look, not sure if we should take Output& in load case or just return it from function

Reply:
I'm usually a fan of using return values but in this case, I see multiple advantages to taking the output by reference:
- It's consistent with the `span` overload
- We can deduce the desired output type (which may be useful if this interface later supports type conversion)

Reply:
Addressed in 349f2592cb7feb996f4590565efa9312ba899773.